### PR TITLE
casper: new package

### PR DESCRIPTION
**Pull request by fanne-stat**
- Added new package casper. Install with gcc@4.8.5. Installing with gcc@7.1.0 causes error.

- Added version 2.2.7 to package jellyfish. Casper depends on jellyfish 2.2.3 or higher.

### CODE REVIEW

File: var/spack/repos/builtin/packages/casper/package.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+       forward and reverse reads."""
+
+    homepage = "http://best.snu.ac.kr/casper/index.php?name=main"
+    url      = "http://best.snu.ac.kr/casper/program/casper_v0.8.2.tar.gz"
+
+    version('0.8.2', '9e83d32ff46b876f33eb1d7b545ec9c2')
+    
+    def install(self, spec, prefix):
+        distutils.dir_util.copy_tree(".", prefix)
+    depends_on('jellyfish@2.2.3:')
```
Comment (davydden) [inline var/spack/repos/builtin/packages/casper/package.py]: we usually put `depends_on` above `install`.

**Comment by akthoma** [comment]
You need to keep the url for version 1.11.1 of jellyfish in the recipe still since the two versions are from different places. Do:

```
version('2.2.7', 'f741192d9061f28e34cb67c86a1027ab')
version('1.1.11', 'dc994ea8b0896156500ea8c648f24846', 
         url='http://www.cbcb.umd.edu/software/jellyfish/jellyfish-1.1.11.tar.gz')
```
where you declare versions. Otherwise it will break the kraken package that specifically depends on `jellyfish@1.11.1`.

**Comment by adamjstewart** [comment]
Minor flake8 error causing Travis to fail:
```
var/spack/repos/builtin/packages/casper/package.py:49: [W291] trailing whitespace
```

**Comment by adamjstewart** [comment]
It looks like you accidentally included hundreds of other commits in your PR. Can you do a clean rebase off of the latest develop?

**Comment by fanne-stat** [comment]
Opened a new PR #6873

**Pull request closed without merging**